rclcpp/test/rclcpp/test_parameter_service.cpp

```diff
@@ -119,3 +120,32 @@ TEST_F(TestParameterService, describe_parameters) {
     EXPECT_EQ(0u, parameter_descs.size());
   }
 }
+
+TEST_F(TestParameterService, parameter_descriptor) {
+  {
+    rclcpp::ParameterDescription param_description;
```

Comment:
I was expecting we were going to have the class `ParameterDescription` templated with `ParameterType` or something else to support syntactic sugar like following? So that it would be easier to create descriptor for the user application? current code almost provides the same experience with `rcl_interfaces::msg::ParameterDescriptor`?

```suggestion
    rclcpp::ParameterDescription<PARAMETER_INTEGER>
    param_description(
      "int_parameter",
      "description",
      "constraints",
      false,
      start, end, step
      );
   node->declare_parameter(rclcpp::ParameterValue{1}, param_description.build());
```

above is just a pseudo cod. i might be mistaken on this, probably we want to get feedback from @wjwwood @gbalke .